tests/v2/validation/observability/rbac/stackstate_rbac_test.go

```diff
@@ -0,0 +1,257 @@
+			Version:     latestUIPluginVersion,
+		}
+
+		err = charts.InstallStackstateExtension(client, extensionOptions)
+		require.NoError(rb.T(), err)
+
+		log.Info("Adding stack state extension configuration.")
+
+		steveAdminClient, err := client.Steve.ProxyDownstream(localCluster)
+		require.NoError(rb.T(), err)
+
+		crdConfig := observability.NewStackstateCRDConfiguration(charts.StackstateNamespace, rb.stackstateConfigs)
+		crd, err := steveAdminClient.SteveType(charts.StackstateCRD).Create(crdConfig)
+		require.NoError(rb.T(), err, "Unable to install stackstate CRD configuration.")
+
+		_, err = steveAdminClient.SteveType(charts.StackstateCRD).ByID(crd.ID)
+		require.NoError(rb.T(), err)
+	}
+
+	latestSSVersion, err := rb.client.Catalog.GetLatestChartVersion(charts.StackstateK8sAgent, rancherPartnerCharts)
+	require.NoError(rb.T(), err)
+	rb.stackstateAgentInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestSSVersion,
+		ProjectID: rb.project.ID,
+	}
+}
+
+func (rb *StackStateRBACTestSuite) TestClusterOwnerInstallStackstate() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := rb.client.WithSession(subSession)
+	require.NoError(rb.T(), err)
+
+	initialStackstateAgent, err := extencharts.GetChartStatus(client, rb.cluster.ID, charts.StackstateNamespace, charts.StackstateK8sAgent)
+	require.NoError(rb.T(), err)
+
+	if initialStackstateAgent.IsAlreadyInstalled {
+		rb.T().Skip("Stack state agent is already installed, skipping the tests.")
+	}
+
+	var newUser *management.User
+	user, err := users.CreateUserWithRole(rb.client, users.UserConfig(), rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+	newUser = user
+	rb.T().Logf("Created user: %v", newUser.Username)
+
+	standardClient, err := rb.client.AsUser(user)
+	require.NoError(rb.T(), err)
+
+	err = users.AddClusterRoleToUser(rb.client, rb.cluster, user, rbac.ClusterOwner.String(), nil)
+	require.NoError(rb.T(), err)
+
+	systemProject, err := projects.GetProjectByName(client, rb.cluster.ID, systemProject)
+	require.NoError(rb.T(), err)
+	systemProjectID := strings.Split(systemProject.ID, ":")[1]
+
+	err = charts.InstallStackstateAgentChart(standardClient, rb.stackstateAgentInstallOptions, rb.stackstateConfigs.ClusterApiKey, rb.stackstateConfigs.Url, systemProjectID)
+	require.NoError(rb.T(), err)
+	log.Info("Stackstate agent chart installed successfully")
+
+	rb.T().Log("Verifying the deployments of stackstate agent chart to have expected number of available replicas")
+	err = extencharts.WatchAndWaitDeployments(client, rb.cluster.ID, charts.StackstateNamespace, meta.ListOptions{})
+	require.NoError(rb.T(), err)
+
+	rb.T().Log("Verifying the daemonsets of stackstate agent chart to have expected number of available replicas nodes")
+	err = extencharts.WatchAndWaitDaemonSets(client, rb.cluster.ID, charts.StackstateNamespace, meta.ListOptions{})
+	require.NoError(rb.T(), err)
+}
```

Comment:
I would add 

```

	podErrors := pods.StatusPods(client, status.ClusterName)
	assert.Empty(t, podErrors)
```

to also check that installing stackstate didn't break something else in the cluster (this is a standard check we usually add in tests like these) 

Side note; do you need to verify that the chart comes up correctly since this is just rbac?

Reply:
ok, will add the pods check. 
as this is part of the chart's check, it’s an important check to make sure that the chart comes up correctly, even for RBAC tests. This way, we can verify that both RBAC permissions and the chart are functioning as expected.